adr/ADR-43.md

```diff
@@ -0,0 +1,102 @@
+| long waitTimeout                            | the timeout when waiting for a publish to be acknowledged. Defaults to 5000ms                                                                                                                                                                                                |
+| PublisherListener publisherListener         | a callback for the user to see what's going on in the workflow, see description of Flight later                                                                                                                                                                              |
+
+
+## PublisherListener Interface
+
+The callback interface for the user to get information about the publish workflow
+
+| Method                                      | Description                                                           |
+|---------------------------------------------|-----------------------------------------------------------------------|
+| void published(Flight flight)               | the flight is ready when the message is published                     |
+| void acked(Flight flight);                  | the publish ack was received                                          |
+| void completedExceptionally(Flight flight); | the publish exceptioned, such as a 503 or lower level request timeout |
+| void timeout(Flight flight)                 | the ack was not returned in time based on waitTimeout                 |
+
+## Flight structure
+
+An object that holds state of the message as it makes its way through the workflow
+
+| Property                                            | Description                                                                          |
+|-----------------------------------------------------|--------------------------------------------------------------------------------------|
+| long getPublishTime()                               | the time when the actual js publish was executed. Used for waitTimeout determination |
+| CompletableFuture<PublishAck> getPublishAckFuture() | the future for the publish ack                                                       |
+| String getId()                                      | `<idPrefix>-<number>` for example                                                    |
+| String getSubject()                                 | the original message subject                                                         |
+| Headers getHeaders()                                | the original message headers                                                         |
+| byte[] getBody()                                    | the original message body                                                            |
+| PublishOptions getOptions()                         | the original message publish options (i.e. expectations)                             |
+
+## Behavior
```

Comment:
The whole behavior description is very Java-specific and does not apply for the most clients.